swim/join_sender.go

```diff
@@ -229,6 +231,10 @@ func (j *joinSender) JoinCluster() ([]string, error) {
 
 	for {
 		if j.node.Destroyed() {
+			j.node.emit(JoinFailedEvent{
```

Comment:
What pattern for event emission are you deciding on? Is there one event emitter per package? Should there be multiple emitters per package? Instead of assuming that the `node` will always be an event emitter, how about passing in an EventEmitter type?